x-pack/plugins/lens/public/shared_components/axis/extent/axis_extent_settings.tsx

```diff
@@ -47,54 +52,12 @@ export function AxisBoundsControl({
     type === 'metric' ? MetricAxisBoundsControl : BucketAxisBoundsControl;
   return (
     <ModeAxisBoundsControl {...props} disableCustomRange={disableCustomRange}>
-      {canShowNiceValues ? (
-        <EuiFormRow
-          label={i18n.translate('xpack.lens.fullExtent.niceValues', {
-            defaultMessage: 'Round to nice values',
-          })}
-          display="columnCompressedSwitch"
-          fullWidth
-        >
-          <EuiSwitch
-            showLabel={false}
-            label={i18n.translate('xpack.lens.fullExtent.niceValues', {
-              defaultMessage: 'Round to nice values',
-            })}
-            data-test-subj={`${testSubjPrefix}_axisExtent_niceValues`}
-            checked={Boolean(extent.niceValues == null || extent.niceValues)}
-            onChange={() => {
-              setExtent({
-                ...extent,
-                mode: modeForNiceValues,
-                niceValues: !Boolean(extent.niceValues == null || extent.niceValues),
-              });
-            }}
-            compressed
-          />
-        </EuiFormRow>
-      ) : null}
       {canShowCustomRanges ? (
         <RangeInputField
-          isInvalid={inclusiveZeroError || boundaryError}
+          isInvalid={Boolean(errorMsg)}
           label={' '}
-          helpText={
-            shouldIncludeZero && (!inclusiveZeroError || boundaryError)
-              ? i18n.translate('xpack.lens.axisExtent.inclusiveZero', {
-                  defaultMessage: 'Bounds must include zero.',
-                })
-              : undefined
-          }
-          error={
-            boundaryError
-              ? i18n.translate('xpack.lens.axisExtent.boundaryError', {
-                  defaultMessage: 'Lower bound has to be larger than upper bound',
-                })
-              : shouldIncludeZero && inclusiveZeroError
-              ? i18n.translate('xpack.lens.axisExtent.inclusiveZero', {
-                  defaultMessage: 'Bounds must include zero.',
-                })
-              : undefined
-          }
+          helpText={helpMsg}
+          error={errorMsg}
```

Comment:
This code was unnecessarily complex to assign error or help messaging. Now the same helper function (`validateExtent`) just returns an error or help message to display. Since we don't care about the type of error it simplifies the `isInvalid` input.